batch/batch/worker/worker_api.py

```diff
@@ -0,0 +1,61 @@
+import abc
+from typing import Dict
+
+from hailtop import httpx
+from hailtop.utils import check_shell, CalledProcessError, sleep_and_backoff
+
+from .disk import CloudDisk
+from .credentials import CloudUserCredentials
+from ..instance_config import InstanceConfig
+
+
+class CloudWorkerAPI(abc.ABC):
```

Comment:
This is really cool and I like it a lot.

Reply:
This was Daniel's addition :)